### ISSUE TITLE
Balance assignment not working with automated transaction

### ISSUE THREAD

**Issue by agarrharr** [ledger/ledger#525]
Here is a very simple example of the problem that I'm seeing. If you get rid of the balance assignment at the end and run: `ledger balance`, then the balance of checking is 0, but if I leave the balance assignment at the end, I get the error that the Unbalanced remainder is $18.00.

```
= /^Expenses:/
        [Assets:Budget:$account]     -1.0
        [Assets:Checking]             1.0

2018/01/02 * Walmart
    Expenses:Food:Groceries             $18.00
    Assets:Checking

2018/01/02 * Zero-based Budget
    [Assets:Checking]              = $0.00
```

So, it seems like the problem is that when it checks the balance assignment, it isn't taking the automated transaction into consideration.

**Comment by jwiegley** [comment]
You're right, this is an error. Something is happening out of sequence.

**Comment by tbm** [comment]
PR 552 said it fixes this issue.  @agarrharr @dedmons can you confirm?

**Comment by tbm** [comment]
or it may have been fixed by

commit 6853464fd0c4a0428540edc635f1257cfb534bcd
Author: Christoph Dittmann <github@christoph-d.de>
Date:   Sat Jun 9 21:57:39 2018 +0100

    Make automated transactions work with assertions
    
    This fixes issue #1127.  In my understanding, PR #552 was meant to fix
    this, but was incomplete.

**Comment by agarrharr** [comment]
What version of ledger is this fix in? I'm running `Ledger 3.1.1-20160111` and it  still has this error.

**Comment by tbm** [comment]
> What version of ledger is this fix in? I'm running Ledger 3.1.1-20160111 and it still has this error.

It's fixed on the `next` branch in Git.  No new version of ledger has been released with the fix yet.

(I believe the version number is meaningless right now since next has the same version afaict.)

**Comment by 501st-alpha1** [comment]
Using the example in the OP, I can verify this is fixed in `Ledger 3.1.3-20190331` (which is the current [Docker image](https://github.com/dcycle/docker-ledger)).

**Comment by agarrharr** [comment]
Yes, this works now in `Ledger 3.1.3-20190331`. Thanks!
